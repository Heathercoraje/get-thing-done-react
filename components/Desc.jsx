import React, { Component } from 'react';
import ReactDOM from 'react';
import styled from 'styled-components';

const DescBox = styled.div`
	margin-bottom: 3vh;
`;

class Desc extends Component {
	render() {
		const name = this.props.name;
		const description = this.props.description;
		return (
			<DescBox className="desc-box">
				<DescForm
					name={name}
					description={description}
					HandleDesc={this.props.HandleDesc}
				/>
			</DescBox>
		);
	}
}

class DescForm extends Component {
	state = {
		edit: false
	};
	componentWillReceiveProps() {
		this.setState({
			edit: false
		});
	}
	toggle = () => {
		const prev = this.state.edit;
		this.setState({
			edit: !prev
		});
		setTimeout(() => {
			this.ref.focus();
		});
	};

	render() {
		const name = this.props.name;
		const description = this.props.description;
		return (
			<div>
				<div className="cate-title">
					<h1 style={{ display: 'inline' }} className="todo-name">
						{name}
					</h1>
					<button
						className="button-edit"
						onClick={() => {
							this.toggle();
						}}
					>
						&nbsp;&#10000;
					</button>
				</div>
				<form
					className={this.state.edit ? 'show' : 'hide'}
					onSubmit={evt => {
						this.props.HandleDesc(evt);
						this.toggle();
					}}
				>
					<input
						ref={c => (this.ref = c)}
						size="22"
						className="input-desc"
						type="text"
						placeholder={description}
						onBlur={evt => this.props.HandleDesc(evt)}
					/>
					<input style={{ display: 'none' }} type="submit" />
				</form>
				<div
					id="desc"
					className={`desc ${this.state.edit ? 'hide' : 'show'}`}
					onClick={this.toggle}
				>
					{description}
				</div>
			</div>
		);
	}
}

export default Desc;
