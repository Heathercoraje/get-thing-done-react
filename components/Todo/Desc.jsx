import React, { Component } from 'react';
import ReactDOM from 'react';
import styled from 'styled-components';

const DescBox = styled.div`
	margin-bottom: 3vh;
`;

class Desc extends Component {
	render() {
		const name = this.props.name;
		const description = this.props.description;
		return (
			<DescBox className="desc-box">
				<DescForm
    name={name}
					description={description}
    HandleDesc={this.props.HandleDesc}
  />
  </DescBox>
		);
	}
}

class DescForm extends Component {
	state = {
		edit: false
	};
	componentWillReceiveProps() {
		this.setState({
			edit: false
		});
	}

	toggle = () => {
		const prev = this.state.edit;
		this.setState({
			edit: !prev
		});
	};

	render() {
		const name = this.props.name;
		const description = this.props.description;
		if (this.state.edit) {
			return (
				<div>
					<Header name={name} toggle={this.toggle} />
					<form
						className="form-desc"
						onSubmit={evt => {
							this.props.HandleDesc(evt);
							this.toggle();
						}}
					>
						<input
							size="22"
							className="input-desc"
							type="text"
							placeholder={description}
							onBlur={evt => this.props.HandleDesc(evt)}
							autoFocus
						/>
						<input style={{ display: 'none' }} type="submit" />
					</form>
				</div>
			);
		} else {
			return (
				<div>
					<Header name={name} toggle={this.toggle} />
					<div id="desc" onClick={this.toggle}>
						{description}
					</div>
				</div>
			);
		}
	}
}

const Header = props => {
	return (
		<div className="cate-title">
			<h1 className="todo-name">{props.name}</h1>
			<button type="button" className="button-edit" onClick={props.toggle}>
				&nbsp;&#10000;
			</button>
		</div>
	);
};

export { Desc, DescForm, Header };
