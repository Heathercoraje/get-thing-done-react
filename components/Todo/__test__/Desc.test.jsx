import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import { Desc, DescForm, Header } from '../Desc';

test('Desc should render correctly', () => {
	const component = shallow(<DescForm />);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
});

test('Check DescForm initial edit state', () => {
	const component = shallow(<DescForm />);
	expect(component.state('edit')).toBe(false);
});
