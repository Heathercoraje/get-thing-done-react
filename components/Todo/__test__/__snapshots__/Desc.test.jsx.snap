// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desc should render correctly 1`] = `
<div>
  <Header
    toggle={[Function]}
  />
  <div
    className="testing"
    id="desc"
    onClick={[Function]}
  />
</div>
`;
